metals/src/main/scala/scala/meta/internal/worksheets/MetalsWorksheetProvider.scala

```diff
@@ -0,0 +1,352 @@
+        ctx.reporter,
+        sectionInputs,
+        instrumented,
+        path.toString
+      )
+      val decorations = for {
+        section <- rendered.sections.iterator
+        statement <- section.section.statements
+      } yield {
+        val pos = statement.position
+        val range = new l.Range(
+          new l.Position(pos.startLine, pos.startColumn),
+          new l.Position(pos.endLine, pos.endColumn)
+        )
+        val margin = math.max(
+          20,
+          userConfig().screenWidth - statement.position.endColumn
+        )
+        val isEmptyValue = isUnitType(statement) || statement.binders.isEmpty
+        val contentText: String = Iterator[Iterator[Char]](
+          commentHeader.iterator, {
+            if (isEmptyValue) {
+              if (statement.out.isEmpty()) "".iterator
+              else statement.out.linesIterator.next().toCharArray().iterator
+            } else {
+              val isSingle = statement.binders.lengthCompare(1) == 0
+              for {
+                (binder, i) <- statement.binders.iterator.zipWithIndex
+                text <- Iterator[Iterator[Char]](
+                  if (isSingle) List.empty[Char].iterator
+                  else {
+                    val comma = if (i == 0) "" else ", "
+                    s"$comma${binder.name}=".iterator
+                  },
+                  pprint.PPrinter.BlackWhite
+                    .tokenize(
+                      binder.value,
+                      width = margin
+                    )
+                    .map(_.getChars)
+                    .filterNot(_.iterator.forall(_.isWhitespace))
+                    .flatMap(_.iterator)
+                    .filter {
+                      case '\n' => false
+                      case _ => true
+                    }
+                    .take(margin)
+                )
+                char <- text
+              } yield char
+            }
+          }
+        ).flatten.mkString
+        val hoverMessage: String = Iterator[Iterator[String]](
+          if (isEmptyValue) {
+            List.empty[String].iterator
+          } else {
+            for {
+              binder <- statement.binders.iterator
+              text <- Iterator(
+                "\n",
+                binder.name,
+                ": ",
+                binder.tpe.render(TPrintColors.BlackWhite),
+                " = "
+              ) ++ pprint.PPrinter.BlackWhite
+                .tokenize(binder.value, width = 100)
+                .map(_.plainText)
+            } yield text
+          },
+          statement.out.linesIterator.flatMap { line =>
+            Iterator("\n// ", line)
+          }
+        ).flatten.mkString
+        DecorationOptions(
+          range,
+          new l.MarkedString("scala", hoverMessage),
+          ThemableDecorationInstanceRenderOptions(
+            after = ThemableDecorationAttachmentRenderOptions(
+              contentText,
+              color = "green",
```

Comment:
It's fine for this first iteration, but I wonder if we can let the editors decide on how to render these.

It feels strange to decide on styling details server-side.

Reply:
I agree, but my understanding is that there's no way for us to access the colors for things like comments https://github.com/Microsoft/vscode/issues/32813

Reply:
We could hardcode this value in the client but I'm not sure that's any better 🤔 Or maybe it is better to send some enum to the client
```scala
object Colors {
  val comment = 1
  val identifier = 2
}
```
and then have the VS Code client translate that into colors

Reply:
My first idea would be to use tmLanguage classes, but then I don’t know how easy is for the client to use them